src/main/java/ai/elimu/dao/jpa/StoryBookChapterDaoJpa.java

```diff
@@ -20,4 +21,16 @@ public List<StoryBookChapter> readAll(StoryBook storyBook) throws DataAccessExce
             .setParameter("storyBook", storyBook)
             .getResultList();
     }
+
+    @Override
+    public List<StoryBookChapter> readAllWithImage(Image image) throws DataAccessException {
+        return em.createQuery(
+                        "SELECT ch " +
+                                "FROM StoryBookChapter ch " +
+                                "WHERE ch.image.id = :image ")
```

Comment:
Should it not be `ch.image` instead of `ch.image.id`? Maybe add a `StoryBookChapterDaoTest.java` to check if the method works?

```suggestion
                                "WHERE ch.image = :image ")
```

Reply:
Two method works but I changed like you suggest because this method using in other places.